src/index.ts

```diff
@@ -50,18 +60,26 @@ export function newConfig<S>(spec: S, env: Environment, options: ConfigOptions =
 }
 
 function logOrFailIfErrors(options: ConfigOptions, errors: Error[], ignoreErrors: boolean) {
-  if (errors.length > 0) {
-    const message = errors.map((e) => e.message).join(", ");
-    if (!ignoreErrors) {
-      throw new ConfigError(message);
-    } else {
-      if (options.doNotLogErrors !== true) {
-        // In theory should use some configurable log library but typically that would
-        // require booting up the very environment/config variables we're trying to resolve.
```

Comment:
I definitely see where this comment is coming from. Since we're following a create-and-memoize pattern, this could pose problems.

For the graphql-service, however, we've got a `createPino()` method that doesn't rely on any environment variables/config, so we should be able to pass a new pino instance to this method (albeit, not one from the `Context` object). I think the tradeoff of seeing error-level logs is worth it, though.